scipy/linalg/tests/test_decomp_cossin.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_almost_equal
+from scipy.stats import ortho_group, unitary_group
+from scipy.linalg import cossin
+
+@pytest.mark.parametrize('m, p, q, dtype, decimals',
+                         [
+                             (3, 2, 1, np.complex128, 12),
+                             (4, 2, 1, np.complex128, 12),
```

Comment:
Probably we won't need all these cases here and I want to still have tests for the error cases and the sunblock case

Reply:
you can nest parametrizations (see the added `test_orcsd_uncsd_lwork` below) then it cross products all cases. The DTYPES variable is used just because of this purpose. And we are missing the complex64 and float32 variables here.

Reply:
Added test cases